File: AVCapturePhotoOutput/AVCapturePhotoOutputObjectiveCSample/AVCapturePhotoOutputObjectiveCSample/MainViewController.m
//
//  MainViewController.m
//  AVCapturePhotoOutputObjectiveCSample
//
//  Created by Nikolay Andonov on 12/6/16.
//  Copyright © 2016 Mentormate. All rights reserved.
//

#import "MainViewController.h"
#import "CapturePhotoHandler.h"

@interface MainViewController () <CapturePhotoHandlerDelegate>

@property (weak, nonatomic) IBOutlet UIBarButtonItem *takePhotoButton;
@property (weak, nonatomic) IBOutlet UIView *cameraOverlayView;

@end

@implementation MainViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [[CapturePhotoHandler sharedHandler] createCaptureSessionForView:self.cameraOverlayView];
    [CapturePhotoHandler sharedHandler].delegate = self;
}

- (void)dealloc {
    
    [CapturePhotoHandler sharedHandler].delegate = nil;
}

- (IBAction)takeLivePhotoButtonAction:(id)sender {
    
    if ([[CapturePhotoHandler sharedHandler] isLivePhotoSupported]) {
        
        [[CapturePhotoHandler sharedHandler] takeLivePhoto];
        self.takePhotoButton.enabled = NO;
    }
    else {
        
        UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Error" message:@"Live Photo capture is not supported on this device" preferredStyle:UIAlertControllerStyleAlert];
        
        UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil];
        [alertController addAction:okAction];
        
        [self presentViewController:alertController animated:YES completion:nil];
    }
}

#pragma mark - CapturePhotoHandlerDelegate

- (void)didEndCapturingLivePhotoWithSuccess:(BOOL)success {
    
    self.takePhotoButton.enabled = YES;
    
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:nil message:@"Live Photo was captured" preferredStyle:UIAlertControllerStyleAlert];
    
    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil];
    [alertController addAction:okAction];
    
    [self presentViewController:alertController animated:YES completion:nil];
    
}


@end

File: AVCapturePhotoOutput/AVCapturePhotoOutputSwiftSample/AVCapturePhotoOutputSwiftSample/MainViewController.swift
//
//  MainViewController.swift
//  AVCapturePhotoOutputSwiftSample
//
//  Created by Nikolay Andonov on 11/11/16.
//  Copyright © 2016 Mentormate. All rights reserved.
//

import UIKit

class MainViewController: UIViewController, CapturePhotoHandlerDelegate {
    
    @IBOutlet weak var takePhotoButton: UIBarButtonItem!
    @IBOutlet weak var cameraOverlayView: UIView!

    override func viewDidLoad() {
        super.viewDidLoad()
        
        CapturePhotoHandler.sharedHandler.createCaptureSession(forView: cameraOverlayView);
        CapturePhotoHandler.sharedHandler.delegate = self
    }

    deinit {
        
        CapturePhotoHandler.sharedHandler.delegate = nil
    }
    
    @IBAction func takeLivePhotoButtonAction(_ sender: Any) {
        
        if CapturePhotoHandler.sharedHandler.isLivePhotoCaptureSupported() {
            
            CapturePhotoHandler.sharedHandler.takeLivePhoto()
            takePhotoButton.isEnabled = false
        }
        else {
            
            let alertController = UIAlertController(title: "Error", message: "Live Photo capture is not supported on this device", preferredStyle: .alert)
            let оKAction = UIAlertAction(title: "ОК", style: .default, handler: nil)
            alertController.addAction(оKAction)
            present(alertController, animated: true, completion: nil)

        }
    }
    
    //MARK - CapturePhotoHandlerDelegate
    
    func didEndCapturingLivePhoto(success: Bool) {
        
        takePhotoButton.isEnabled = true
        
        let alertController = UIAlertController(title: "", message: "Live Photo was captured!", preferredStyle: .alert)
        let оKAction = UIAlertAction(title: "ОК", style: .default, handler: nil)
        alertController.addAction(оKAction)
        present(alertController, animated: true, completion: nil)
    }
}
